src/backend/joanie/badges/models.py

```diff
@@ -0,0 +1,166 @@
+"""
+Declare and configure the models for the badges part
+"""
+import uuid
+from functools import lru_cache
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from ..core.models import User
+
+
+@lru_cache
+def get_badge_provider_choices():
+    """Get BadgeProvider choices. Should be evaluated once."""
+    # pylint: disable=fixme
+
+    # FIXME
+    # Dynamically list available providers
+    choices = ("FAK", "OBF")
+    return models.TextChoices("BadgeProvider", choices)
+
+
+class Badge(models.Model):
```

Comment:
Don't know what is the best, but for similar models (related to certification) we are using another naming convention: 
- `CertificateDefinition` to store issuable certificate
- `Certificate` to store issued certificate

I like also your naming convention but I think we should use the same naming convention across those 2 models for consistency.

Reply:
I've used OBF semantic. I am afraid that following your convention, we will introduce a confusion by using `Badge` for an issued badge.